src/identity/components/GlobalHeader/IdentityUserAvatar.tsx

```diff
@@ -0,0 +1,131 @@
+import React, {CSSProperties} from 'react'
+import {IdentityUser} from 'src/client/unityRoutes'
+import {
+  Button,
+  ButtonShape,
+  ClickOutside,
+  ComponentColor,
+  FlexBox,
+  Icon,
+  IconFont,
+  InfluxColors,
+} from '@influxdata/clockface'
+
+import './UserPopoverStyles.scss'
+import {Link} from 'react-router-dom'
+
+type Props = {
+  user: IdentityUser
+}
+
+type State = {
+  isPopoverOpen: boolean
+}
+
+class IdentityUserAvatar extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      isPopoverOpen: false,
+    }
+  }
+
+  private getInitials = (): string => {
+    const {user} = this.props
+    const firstName = user.firstName
+    const lastName = user.lastName
+    const initials = `${firstName.charAt(0)}${lastName.charAt(0)}`
+    return initials
+  }
+
+  // get button style based on whether popover is open or not
+  private getButtonStyle = (): CSSProperties => {
+    const {isPopoverOpen} = this.state
+
+    const style: CSSProperties = {
+      margin: 0,
+      borderRadius: '50%',
+      border: isPopoverOpen ? 'none' : `2px solid ${InfluxColors.Grey65}`,
+      background: isPopoverOpen
+        ? `linear-gradient(75.66deg, #0098F0 0%, ${InfluxColors.Amethyst} 79.64%)`
+        : 'none',
+    }
+    return style
+  }
+
+  private togglePopoverState = () => {
+    const {isPopoverOpen} = this.state
+    this.setState({isPopoverOpen: !isPopoverOpen})
+  }
+
+  private setPopoverStateClosed = () => {
+    this.setState({isPopoverOpen: false})
+  }
+
+  private getUserPopoverContents = () => {
+    const {user} = this.props
+    return (
+      <div className="user-popover">
+        <div className="user-popover-header">
+          <div className="user-popover-header-name">
+            {user.firstName} {user.lastName}
+          </div>
+          <div className="user-popover-header-email">{user.email}</div>
+          <hr />
+        </div>
+        <div className="user-popover-footer">
+          <Link className="user-popover-footer--button" to="/">
```

Comment:
Looks good. We can decide which URL we want the link to point to once UserProfile page is done.

Reply:
https://github.com/influxdata/ui/issues/5111